examples/search_examples/public/components/app.tsx

```diff
@@ -0,0 +1,343 @@
+        },
+      });
+  };
+
+  const onClickHandler = () => {
+    doAsyncSearch();
+  };
+
+  const onMyStrategyClickHandler = () => {
+    doAsyncSearch('myStrategy');
+  };
+
+  const onServerClickHandler = async () => {
+    if (!indexPattern || !selectedField) return;
+    try {
+      const response = await http.get(SERVER_SEARCH_ROUTE_PATH, {
+        query: {
+          index: indexPattern.title,
+          field: selectedField.name,
+        },
+      });
+
+      notifications.toasts.addSuccess(`Server returned ${JSON.stringify(response)}`);
+    } catch (e) {
+      notifications.toasts.addDanger('Failed to run search');
+    }
+  };
+
+  if (!indexPattern) return null;
+
+  return (
+    <Router basename={basename}>
+      <I18nProvider>
+        <>
+          <navigation.ui.TopNavMenu
+            appName={PLUGIN_ID}
+            showSearchBar={true}
+            useDefaultBehaviors={true}
+            indexPatterns={indexPattern ? [indexPattern] : undefined}
+          />
+          <EuiPage restrictWidth="1000px">
+            <EuiPageBody>
+              <EuiPageHeader>
+                <EuiTitle size="l">
+                  <h1>
+                    <FormattedMessage
+                      id="searchExamples.helloWorldText"
+                      defaultMessage="{name}"
+                      values={{ name: PLUGIN_NAME }}
+                    />
+                  </h1>
+                </EuiTitle>
+              </EuiPageHeader>
+              <EuiPageContent>
+                <EuiPageContentBody>
+                  <EuiText>
+                    <EuiFlexGrid columns={1}>
+                      <EuiFlexItem>
+                        <EuiFormLabel>Index Pattern</EuiFormLabel>
+                        <IndexPatternSelect
+                          savedObjectsClient={savedObjectsClient}
+                          placeholder={i18n.translate(
+                            'backgroundSessionExample.selectIndexPatternPlaceholder',
+                            {
+                              defaultMessage: 'Select index pattern',
+                            }
+                          )}
+                          indexPatternId={indexPattern?.id || ''}
+                          onChange={async (newIndexPatternId: any) => {
+                            const newIndexPattern = await data.indexPatterns.get(newIndexPatternId);
+                            setIndexPattern(newIndexPattern);
+                          }}
+                          isClearable={false}
+                        />
+                      </EuiFlexItem>
+                      <EuiFlexItem>
+                        <EuiFormLabel>Numeric Fields</EuiFormLabel>
+                        <EuiComboBox
+                          options={formatFieldsToComboBox(numericFields)}
+                          selectedOptions={formatFieldToComboBox(selectedField)}
+                          singleSelection={true}
+                          onChange={(option: any) => {
```

Comment:
Nit: I think this type is inferred and doesn't need `any`.

```suggestion
                          onChange={(option) => {
```